### PR TITLE
fixes #9402 - allow removal of cv version with archive deletion

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
alternative implementation to https://github.com/Katello/katello/pull/5015

After speaking with ivan, i think this maybe superior.  This changes the ContentViewVersion::Destroy action to only destroy archived repos/puppet env, and relaxes the check in the case of

**Comment by jlsherrill** [comment]
[test]

File: app/lib/actions/katello/content_view/remove.rb [hunk 1]
```diff
@@ -74,6 +75,18 @@ def plan(content_view, options)
           end
         end
 
+        def check_version_deletion(versions, cv_envs)
+          versions.each do |version|
+            unless version.environments.empty?
```
Comment (ehelms) [inline app/lib/actions/katello/content_view/remove.rb]: Do you need this check, given that if `version.environments` is empty, nothing inside the `each` statement will be run?

File: app/lib/actions/katello/content_view/remove.rb [hunk 2]
```diff
@@ -74,6 +75,18 @@ def plan(content_view, options)
...
+              version.environments.each do |env|
+                if cv_envs.select { |cv_env| cv_env.content_view_version == version && cv_env.environment == env }.empty?
```
Comment (ehelms) [inline app/lib/actions/katello/content_view/remove.rb]: I think you could use `none?` here.

**Comment by ehelms** [comment]
I am getting a JS error on the verion page after I start the delete -- not sure if it directly related to your change or not.

**Comment by jlsherrill** [comment]
@ehelms js error was unrelated, but i fixed it

**Comment by ehelms** [comment]
ACK

**Pull request merged**